pkg/auth/policy.go

```diff
@@ -0,0 +1,268 @@
+package auth
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"golang.org/x/crypto/openpgp"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+// A Policy encapsulates the behavior a p2 node needs to authorize
+// its actions. It is possible for implementations to rely on other
+// services for these behaviors, so these calls may be slow or
+// transiently fail.
+type Policy interface {
+	// Check if a pod is authorized to be installed and run on this
+	// node. This involves checking that the pod has a valid signature
+	// and that the signer is authorized to install/run the pod. If
+	// the action is authorized, `nil` will be returned.
+	AuthorizePod(
+		id string,
```

Comment:
temp files are just for the artifact signatures. sooner or later they have to be downloaded, but they don't have to be downloaded there. The reason I actually put that code there was because 1) I wasn't really convinced that the verification logic belonged in hoist.launchable and 2) I didn't want to pass the keyring into the launchable (which would have been necessary to verify the signature of the launchable's digest).

if the auth logic is factored out to some third party (as this pr is doing), then sure move the downloading somewhere else too